application/common/core.cpp

```diff
@@ -180,6 +181,39 @@ void loop()
         {
             IMEDelegateDispatcher::shared()->dispatchShowComposition(event.text.text, strlen(event.text.text), event.edit.start, event.edit.length);
         }
+        else if (event.type == SDL_CONTROLLERBUTTONDOWN)
+        {
+            //lmLogInfo(coreLogGroup, "Controller Button Down %d %d %d", event.cbutton.which, event.cbutton.button);
+            LoomGameController::getGameController(LoomGameController::getControllerIndex(event.cbutton.which))->buttonDown(event);
```

Comment:
When a controller connects it gets opened and closed when it disconnects. Only opened controllers trigger events. I do not see a situation where it could fail.

I could add a check to see if a valid object is returned, but I think it is not needed.